### DISCUSSION TITLE
VS Code reports error from tsconfig.json

### DISCUSSION THREAD

**Discussion by zgoda** [preactjs/wmr#515] [Q&A] [answered]
As soon as I added a component module in my project the VS Code started reporting error coming from TS misconfiguration:

```
File '/home/jarek/work/breadcalc/node_modules/source-map/source-map' not found.
  The file is in the program because:
    Root file specified for compilation
```

![Zrzut ekranu z 2021-04-01 17-13-33](https://user-images.githubusercontent.com/141999/113315853-03f2c880-930e-11eb-90b0-67051ac00aba.png)

I tried to install this package but this had no effect.

This is plain JS project, so in dev mode everything works fine so for now it's just annoying. I believe this comes from VS Code built-in typescript processor(?) so I don't know exactly what version is this.

Any suggested workarounds?

**Comment by rschristian** [comment]
Do you still see this in wmr `1.4.2`?

**Reply by zgoda** [reply]
I see this in 1.5.1.

**Reply by rschristian** [reply]
Alright, but do you see it in `1.4.2`? I'm trying to see if it was something introduced with one of the added features in `1.5.0`. I don't use VSCode so I have no clue what it's doing.

**Reply by zgoda** [reply]
I'd say it's something with VS Code.

I was able to trigger this in 1.4.2 using convoluted commands combo. After cleaning fresh wmr project add a component module, import it in index.js and use in App component, then open freshly created tsconfig.json, run "Format document" with editorconfig setting different than file default (mine is 2 spaces), hit "Save", boom. Sometimes goes away right after reloading editor window, sometimes requires VSC restart. And then reappears at random.

I'm trying to confirm/verify before I file a ticket for VSC.

**Comment by zgoda** [comment, marked as answer]
[Issue reported in VSC](https://github.com/microsoft/vscode/issues/132531)

**Reply by zgoda** [reply]
And a [cause](https://github.com/microsoft/TypeScript/issues/42840#issuecomment-945340763) for this has been identified. Neither VSC, TS nor WMR is doing anything wrong here.
